### PR TITLE
fixup: fix conda not syncing conda extension files

### PR DESCRIPTION
**Pull request by jhan217**
Thanks taking the time to contribute. This template should help guide you through the process of creating a pull request for review. Please erase any part of this template that is not relevant to your pull request:

- [x] Have you followed the guidelines for [contributors](http://docs.libraries.io/contributorshandbook)?
- [x] Have you checked to ensure there aren't other open pull requests on the repository for a similar change?
- [x] Is there a corresponding ticket for your pull request?
- [x] Have you written new tests for your changes?
- [x] Have you successfully run the project with your changes locally?

If so then please replace this section with a link to the ticket(s) it addressed, an explanation of your change and why you think we should include it. Thanks again! 

### Summary of changes
The parser was previously only parsing packages & versions from the "packages" subdirectory (which appears to contain all the .bz2 distributions) so it missed all the packages & versions that were in the "packages.conda" subdirectory (which appears to contain all the .conda distributions). 

Unrelatedly, I did notice that there were a few packages that did not have any channel data. Not sure what the actual volume / packages affected are, but I did add a logger for them. 

This PR also fixes a couple of other bugs I noticed while testing, as well as the specs.

### CODE REVIEW

File: channel.rb [hunk 1]
```diff
@@ -45,22 +45,38 @@ def retrieve_packages
...
+        parsed_response = JSON.parse(response.body)
+
+        blobs = [parsed_response["packages"], parsed_response["packages.conda"]]
+
+        blobs.each do |blob|
+          next if blob.nil?
 
-          packages[package_name][:versions] << release_version(key, version)
+          blob.each_key do |key|
+            version = blob[key]
```
Comment (tiegz) [inline channel.rb]:
```suggestion
          blob.each_pair do |artifact_filename, version|
```

File: channel.rb [hunk 2]
```diff
@@ -45,22 +45,38 @@ def retrieve_packages
...
+              puts "Unable to get channel data for package #{package_name} from url: #{repo_data_url}"
+              next
+            end
+
+            unless packages.key?(package_name)
+              package_data = channeldata[package_name]
+              packages[package_name] = base_package(package_data, package_name)
+            end
+
+            packages[package_name][:versions] << release_version(key, version)
```
Comment (tiegz) [inline channel.rb]: and a followup to the prev suggestion: 

```suggestion
            packages[package_name][:versions] << release_version(artifact_filename, version)
```

**Review by macowie — APPROVED** [review]

**Review by jhan217 — COMMENTED** [review]
Added WebMock to stub the requests properly to fix the specs.

**Review by tiegz — APPROVED** [review]
nice debugging!

File: spec/channel_spec.rb [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    end
+
+    it "returns information about a package + version distributed in .conda format" do
+      actual = channel.package_version(package_name, "2.14.3")
+
+      expect(actual.first).to include(
+        artifact: "ansible-core-2.14.3-pyhd8ed1ab_0.conda",
+        download_url: "https://conda.anaconda.org/conda-forge/noarch/ansible-core-2.14.3-pyhd8ed1ab_0.conda",
+        original_license: "GPL-3.0-or-later",
+        published_at: Time.new(2023, 0o2, 27, 23, 47, 26),
```
Comment (macowie) [inline spec/channel_spec.rb]: These assertions (this and L140) fail running in my timezone 🕐 
```
       -:published_at => 2023-02-27 23:47:26.000000000 -0600,
       +:published_at => 2023-02-27 22:47:26.000000000 -0600,
```

**Review by macowie — APPROVED** [review]

**Pull request merged**